Consolidate cost computation in a single executable

abailly-iohk opened the pull request:
* Move hashing cost computation out of ContractSpec into
`contract-cost` executbale
* Move exe to hydra-node because of dependencies

There's probably more refactoring to do in that direction:
* Add CBOR encoding cost computation?
* Factor out common code handling execution costs

abailly committed:
Consolidate cost computation in a single executable

* Move hashing cost computation out of ContractSpec into
`contract-cost` executbale
* Move exe to hydra-node because of dependencies

ch1bo reviewed (approved):
:nail_care: :+1:

ghost commented:
OK, build is failing because of missing GHC flags because of Plutus compiler...

abailly committed:
Added flags to help haddock overcome plutus hurdle

abailly committed:
Add hydra-development flag to tx-cost which depends on plutus thing

abailly committed:
Add -haddock ghc flag and remove defer error

Aligning with what's in the plutus-cbor project

abailly committed:
[test] Remove plutus-merkle-tree package

abailly committed:
Forgot test run in build file

KtorZ reviewed (approved):
Possible follow-ups:

- Also move the plutus-cbor evaluation costs to this executable.
- Possibly, show plots using `gnuplot` or at least, construct some gnuplot drop-in configuration (so set axis, plot's name, etc..)

abailly committed:
Revert "Forgot test run in build file"

This reverts commit abe33493391b89f4d5d700524e5eafced52ecf76.

abailly committed:
Revert "[test] Remove plutus-merkle-tree package"

This reverts commit 3e343c37323cfbe0b6a3ae54147c8969127ffdfa.

abailly committed:
Disable hoogle when running CI

abailly committed:
Add missing semicolon

abailly committed:
Implement CI-only nix shell without hoogle

abailly committed:
Disable hoogle for the sake of testing CI

abailly committed:
Remove ci attribute which does not work

abailly committed:
Fix compilation errors

abailly committed:
Move MT validators to tx-cost exe component

Haddock insists on failing to work when the Validators are in the library, so trying to move
them where they are actually used.

abailly committed:
Revert all changes to nix stuff

abailly committed:
Revert "Move MT validators to tx-cost exe component"

This reverts commit 3513850998761d5a03cf1c23d71bbb6d2469d6db.

abailly committed:
Defer plugin error for MT library when flag set

abailly committed:
Remove offending validator

CI Build chokes on this at nix build stage because it tries to build
haddock for Hoogle use. It seems the error might be tied to the
emptyValidator so checking it's indeed the case.

abailly committed:
Configure haddock-options on plutus projects

This ensure plugin errors are ignored when generating haddock for code
that uses Plutus plugin

abailly committed:
Really move validators out of plutus-merkle-tree package

Those validators are breaking CI when entering nix-shell and we could
not find a way to fix that, so resorting to moving them out of the
way.
